File: 169-majority-element/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 0;
        int candidate = 0;
        int n = nums.size();

        for (int i = 0; i < n; i++){
            if (count == 0){
                candidate = nums[i];
                count += 1;
            } else if (candidate == nums[i]){
                count += 1;
            } else {
                count -=1;
            }
        }
        
        return candidate;
    }
};

File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        candidate = 0
        count = 0

        for i in nums:
            if count == 0:
                candidate = i
            
            if candidate == i:
                count+= 1
            else:
                count -=1
            
            
        return candidate